src/Module/Admin/Storage.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2021, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Admin;
+
+use Friendica\Core\Renderer;
+use Friendica\DI;
+use Friendica\Model\Storage\IStorage;
+use Friendica\Module\BaseAdmin;
+use Friendica\Util\Strings;
+
+class Storage extends BaseAdmin
+{
+	public static function post(array $parameters = [])
+	{
+		self::checkAdminAccess();
+
+		self::checkFormSecurityTokenRedirectOnError('/admin/storage', 'admin_storage');
+
+		$storagebackend = Strings::escapeTags(trim($_POST['storagebackend'] ?? ''));
+		/** @var IStorage $newstorage */
+		$newstorage = DI::storageManager()->getByName($storagebackend);
+
+		// save storage backend form
+		$storage_opts        = $newstorage->getOptions();
```

Comment:
Isn't this saving only the options of the selected storage method?

Reply:
yes, but it's for the value mapping from the form to the right keys at the later code.

Reply:
So what happens if you update the S3 storage options while "Filesystem" is selected as the current storage backend?

Reply:
hmm ... they won't get saved until you select "S3" .. But on the other hand how can I check for mandatory fields in case S3 is possible but I don't want to configure it ..

So I would need a "save" for each storage type and an "activate" for the specific storage type .. I was at that point already ;-)

Reply:
Excellent, I'm waiting for the changes.

Reply:
I see

Reply:
Done, see screenshot at description :)